src/commands/shane/data/file/download.ts

```diff
@@ -0,0 +1,93 @@
+import { flags, SfdxCommand } from '@salesforce/command';
+import * as fs from 'fs-extra';
+import { Record } from '../../../../shared/typeDefs';
+
+interface ContentVersion extends Record {
+    Title: string;
+    FileExtension: string;
+    VersionData: string;
+}
+interface ContentDocument extends Record {
+    LatestPublishedVersionId: string;
+}
+
+export default class Upload extends SfdxCommand {
+    public static description = 'save a file from the org to the local filesystem';
+
+    public static examples = [
+        `sfdx shane:data:file:download -d 0691k000000MXfkAAG -o ./files/
+    //save a ContentDocument from the org to the files directory, keeping the existing filename`,
+        `sfdx shane:data:file:download -d 0691k000000MXfkAAG -o ./files/King.jpg
+    //save a ContentDocument from the org to files/King.jpg`,
+        `sfdx shane:data:file:download -d 0691k000000MXfkAAG
+    //save a ContentDocument from the org to the current working directory, keeping the existing filename`,
+        `sfdx shane:data:file:download -v 0681k000000MXfkAAG -o ./files/King.jpg
+    //save a ContentVersion from the org to files/King.jpg`
+    ];
+
+    protected static flagsConfig = {
+        documentid: flags.id({ char: 'd', description: 'optional ContentDocument ID that should be downloaded' }),
+        documentversionid: flags.id({ char: 'v', description: 'optional ContentVersion ID that should be downloaded' }),
+        outputpath: flags.filepath({
+            char: 'o',
+            description:
+                'optional output path to save the file, if ommitted will save to current directory, ' +
+                'if directory then it will keep the filename and save into that directory'
+        })
+    };
+
+    // Comment this out if your command does not require an org username
+    protected static requiresUsername = true;
+
+    // tslint:disable-next-line:no-any
+    public async run(): Promise<any> {
+        return new Promise(async (resolve, reject) => {
+            if (this.flags.documentid && this.flags.documentversionid) {
```

Comment:
I prefer the exclusive property on flags: https://oclif.io/docs/flags

Reply:
Never knew about that property, that is exactly what we needed.